README.md

```diff
@@ -20,13 +21,92 @@ from scratch, so you have to start somewhere...
+
+Invocation:
+
+```sh
+$ builder-init [flags] <archetype>
+```
+
+Flags:
+
+```
+  --help
+  --version
+  --prompts
+```
+
+Examples:
+
+```sh
+$ builder-init builder-react-component
+$ builder-init builder-react-component@0.2.0
+$ builder-init FormidableLabs/builder-react-component
+$ builder-init FormidableLabs/builder-react-component#v0.2.0
+$ builder-init git+ssh://git@github.com:FormidableLabs/builder-react-component.git
+$ builder-init git+ssh://git@github.com:FormidableLabs/builder-react-component.git#v0.2.0
+$ builder-init /FULL/PATH/TO/builder-react-component
+```
+
+Internally, `builder-init` utilizes [`npm pack`](https://docs.npmjs.com/cli/pack)
+to download (but not install) an archetype package from npm, GitHub, file, etc.
+There is a slight performance penalty for things like local files which have to
+be compressed and then expanded again, but we gain the very nice benefit of
+allowing `builder-init` to install anything `npm` can in exactly the same
+manner that `npm` does.
+
+### Installing from a Relative Path on the Local Filesystem
+
+One exception to the "install like `npm` does" rule is installation from the
+**local filesystem**. Internally, `builder-init` creates a temporary directory
+to expand the download from `npm pack` and executes the process in that
+directory, meaning that relative paths to a target archetype are now incorrect.
+
+Accordingly, if you _want_ to simulate a relative path install, you can try
+something like:
+
+```sh
+# Mac / Linux
+$ builder-init "${PWD}/../builder-react-component"
+
+# Windows
+$ builder-init "%cd%\..\builder-react-component"
 ```
-TODO: Add usage, documentation.
-https://github.com/FormidableLabs/builder-init/issues/6
+
+### Automating Prompts
+
+To facilitate automation, notably testing an archetype by generating a project
+with `builder-init` and running the project's tests as part of CI, there is a
+special `--prompts=JSON_OBJECT` flag that skips the actual input prompts and
+injects fields straight from a JSON object.
+
+```sh
+$ builder-init <archetype> \
+  --prompts'{"name":"bob","quest":"popcorn","destination":"my-project"}'
 ```
 
+Note that _all_ required fields must be provided in the JSON object, no defaults
+are used, and the init process will fail if there are any missing fields.
+**Tip**: You will need a `destination` value, which is added to all prompts.
+
+A working example is available at:
+[`builder-react-component/.travis.yml`](https://github.com/FormidableLabs/builder-react-component/blob/master/.travis.yml)
+which initializes the archetype's templates for a fresh project with canned
+`--prompts` values, npm installs dependencies, then runs the same `builder`
+tasks used in the project's CI.
+
+
+## Archetype Templates
 
-## Templates
+Authoring a templates for an archetype consists of adding the following to your
+archetype source:
+
+* **`init.js`**: A control file for user prompts and data. See, e.g.,
```

Comment:
This is also unclear, please add more info on `init.js` and `prompts, destination, derived`. What parts are required, what are valid options, invalid values, how are they used, etc.